clientapi/routing/membership.go

```diff
@@ -215,3 +209,41 @@ func getMembershipStateKey(
 
 	return
 }
+
+func checkAndProcessThreepid(
+	req *http.Request,
+	device *authtypes.Device,
+	body *threepid.MembershipRequest,
+	cfg config.Dendrite,
```

Comment:
It seems a lot of places are passing `cfg` not as a pointer. Maybe they need to be changed at some point?

Reply:
Mm, indeed. I don't believe there's any reason not too, and there's RAM savings in doing so.

Reply:
https://github.com/matrix-org/dendrite/issues/736